Framework/Algorithms/src/RunCombinationHelpers/RunCombinationHelper.cpp

```diff
@@ -85,5 +86,72 @@ RunCombinationHelper::checkCompatibility(MatrixWorkspace_sptr ws,
   return errors;
 }
 
+/// @cond
+// Local function used within validateInputWorkspaces() below in a call to
+// std::list::sort(compare) to order the input workspaces by the start of their
+// frame (i.e. the first X value).
+static bool compare(MatrixWorkspace_sptr first, MatrixWorkspace_sptr second) {
+  return (first->x(0).front() < second->x(0).front());
+}
+/// @endcond
+
+/** Checks that the input workspace all exist, that they are the same size, have
+ * the same units
+ *  and the same instrument name. Will throw if they don't.
+ *  @param  inputWorkspaces The names of the input workspaces
+ *  @param g_log A logger to send error messages to
+ *  @return A list of pointers to the input workspace, ordered by increasing
+ * frame starting point
+ *  @throw  Exception::NotFoundError If an input workspace doesn't exist
+ *  @throw  std::invalid_argument    If the input workspaces are not compatible
+ */
+std::list<API::MatrixWorkspace_sptr>
+RunCombinationHelper::validateInputWorkspaces(
+    const std::vector<std::string> &inputWorkspaces, Logger &g_log) {
+  std::list<MatrixWorkspace_sptr> inWS;
+
+  for (size_t i = 0; i < inputWorkspaces.size(); ++i) {
+    MatrixWorkspace_sptr ws;
+    // Fetch the next input workspace - throw an error if it's not there
+    try {
+      ws = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+          inputWorkspaces[i]);
+      if (!ws) {
+        g_log.error() << "Input workspace " << inputWorkspaces[i]
+                      << " not found.\n";
+        throw Kernel::Exception::NotFoundError("Data Object",
+                                               inputWorkspaces[i]);
+      }
+      inWS.push_back(ws);
+    } catch (Exception::NotFoundError &) {
```

Comment:
I understand that it's a cut and paste edit but while we are on it can we improve things a little?

At the moment if I pass a name of an existing table workspace to `SumOverlappingTubes` I get this list of errors:

```
Input workspace Table1 not found.
Input workspace Table1 not found.
Error in execution of algorithm SumOverlappingTubes:
Data Object search object Table1
```
The first message is from the `if (!ws) {...` block, the second one is from the `catch` clause and the third is logged by the Algorithm base class when the exception is re-thrown. I think logging in this method is unnecessary and the message is wrong: the workspace exists but its type is wrong.

Reply:
A good point. I will remove the try/catch as I don't think it is required, it should throw from `retrieveWS`, but in most cases it will be caught on input validation.